### ISSUE TITLE
Anomalous backslash in string should report the line within multi-line strings

### ISSUE THREAD

**Issue by gpshead** [PyCQA/pylint#3653]
pylint when run on something like this:

```python
SCRIPT = """\
blah
# Don't have blah? Run this command:
# prepare_blah --some_regex=\.[Ss]pam
blah
"""
```

Will report `W1401: Anomalous backslash in string: '\.'. String constant might be missing an r prefix. (anomalous-backslash-in-string)`

on the first line of the multi-line string.

This is extra confusing when lint errors are surfaces with code snippets or inline on the reported line in an editor or code review tool as the reader immediately assumes it is talking about the `"""\` right there (it isn't).

Error reporting for contents of multi-line strings should at least use the line number of the first offence detected within the string to lead the developer to the right place with the issue.  _(or when reporting ranges of lines and offsets, the first and last such offsets within the string)_

In the above example, the \ was buried much further down within the multi-line string content mid-line. The larger the multi-line constant the less obvious this becomes.
